### DISCUSSION TITLE
Custom module with nushell

### DISCUSSION THREAD

**Discussion by noahfraiture** [starship/starship#6444] [Q&A]
Hello,

I've been trying to add a custom module for 2 hours now and I don't find the solution. I use nushell but I don't think it's the issue because it works well in pure command line without starship. 
At some point,  an environment variable will be created, and the module should display its content. But I want to have some control on that data, to reduce its size for example.

This is what I have for now :
```
shell = ["nu", "-c"]
command = "let present = \"PKGS\" in $env; echo $present"
when = "use std assert; let present = 'PKGS' in $env; assert true"
```

This should simply give 'true' or 'false' depending if a set the environment variable or not. These command work in my terminal. However if I manually set the environment variable, nothing change and I still get 'false'. 
Is there something am I missing ? Like something to trigger the change ?

**Comment by MrPandir** [comment]
If this is still useful to you, here is the solution.
If you use the format from [Prompt](https://starship.rs/config/#prompt), don't forget to add it as `$custom` or `${custom.test}`.

```
[custom.test]
command = "'PKGS' in $env"
when = true
shell = ["nu"]
```

<img width="1792" height="1136" src="https://github.com/user-attachments/assets/3bc28958-7c60-43f5-8349-282725936601" />
